src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -525,22 +525,50 @@ extern "C" int32_t CryptoNative_SslAddExtraChainCert(SSL* ssl, X509* x509)
     return 0;
 }
 
-extern "C" int32_t CryptoNative_SslSelectNextProto(unsigned char** out, unsigned char* outlen, const unsigned char* server, unsigned int server_len, const unsigned char* client, unsigned int client_len)
+extern "C" int32_t CryptoNative_SslSelectNextProto(uint8_t** out, uint8_t* outlen, const uint8_t* server, uint32_t server_len, const uint8_t* client, uint32_t client_len)
 {
-    return SSL_select_next_proto(out, outlen, server, server_len, client, client_len);
+#ifdef HAVE_OPENSSL_ALPN
+    if (API_EXISTS(SSL_select_next_proto))
+    {
+        return SSL_select_next_proto(out, outlen, server, server_len, client, client_len);
+    }
+    else
```

Comment:
Technically the else keyword and else braces are redundant, since the method either \#defines out or hits an if+return.

But this is fine if you like it.

Reply:
This is the code pattern used elsewhere in the crypto shim.